File: 11.container-with-most-water.js
// @leet start
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function(height) {
	var l = 0;
	var r = height.length-1;
	var answer =0;
	while (l<r) {
		answer = Math.max(answer, (r-l)*Math.min(height[r],height[l]));
		if (height[l]<height[r]) {
			l++;
		}else {
			r--;
		}
	}
	return answer;
    
};
// @leet end

File: 11.container-with-most-water.cpp
// @leet start
#include <algorithm>
#include <climits>
#include <vector>
using namespace std;
class Solution {
public:
  int maxArea(vector<int> &height) {
    int l = 0;
    int r = height.size()-1;
    int answer = 0;
    while (l < r) {
      answer = max(answer, (r - l) * min(height[l], height[r]) );
      if (height[l] < height[r]) {
        l++;
      }else {
      	r--;
      }
    }
		return answer;
  }
};
// @leet end
